src/libstd/net/addr.rs

```diff
@@ -632,7 +746,9 @@ pub trait ToSocketAddrs {
     ///
     /// # Errors
     ///
-    /// Any errors encountered during resolution will be returned as an `Err`.
+    /// Any errors encountered during resolution will be returned as an [`Err`].
```

Comment:
So yes, agreed with both: this does feel a little redundant right now, but also, we do want to start describing which errors actually can happen. 

You could also just remove this section for the purposes of this PR.